active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,572 @@
+isn't the type's default (for example an int-typed property that doesn't default to zero.) -->
+
+<!-- Option 2: Put these descriptions in the below API Details section,
+with a "///" comment above the member or type. -->
+
+# API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+## InputPropertyAttribute
+The `InputPropertyAttribute` helps us ensure that we only generate validation code for the property specified by this attribute. This will help reduce code bloat by not generating code for properties that will never perform validation (like width). Controls can specify multiple properties as their input property. 
+
+    /// <summary>
+    /// Name of the property that is associated with user input.
+    /// </summary>
+    attribute InputPropertyAttribute
+    {
+         string Name;
+    }
+
+## InputValidationKind  
+Determines how the control displays validation visuals.
+
+    /// <summary>
+    ///    Determines how the control displays it's validation visuals
+    /// </summary>
+    [webhosthidden]
+    enum InputValidationKind
+    {
+        /// <summary>
+        /// Let's the Control dictate how to display visuals. This is the default value.
+        /// </summary>
+        Auto,
+        /// <summary>
+        /// An error icon displays to the right of the control that displays the error messages in a tool tip.
+        /// </summary>
+        Compact,
+        /// <summary>
+        /// Text for the error messages is displayed underneath the control.  
+        /// </summary>
+        Inline,
+        /// <summary>
+        /// Validation visuals are not displayed at all.
+        /// </summary>
+        Hidden,
+    };
+
+## IInputValidationControl
+Any control that wants to participate in input validation should derive from this interface. This, among other things, is what allows the markup compiler to know when to generate the code necessary for listening to the `INotifyDataErrorInfo.ErrorsChanged` event and propagate the results to the `IInputValidationControl.Errors` property.
+
+When the `ValidationCommand` property is non-null, the following will be true:
+1. When the `InputValidationKind` property is set to `Hidden`, the control will *not* invoke the `ValidationCommand.CanValidate` or `ValidationCommand.Validate` methods
+2. When the `InputValidationKind` property is set to `Auto`, the control will use the value of the `ValidationCommand.InputValidationKind` property for the value. If both are `Auto` the default bevahior should be `Inline`.
+3. When focus has been lost, validation should be attempted by calling the `ValidationCommand.CanValidate` method.
+4. If the `IInputValidationControl.ValidationErrors` collection is non-empty, validation should be attempted by calling the `ValidationCommand.CanValidate` method.
+
+The markup compiler uses this interface to properly generate code when using x:Bind iff:
```

Comment:
"iff" - typo?

Reply:
Or if means "if and only if", please write out the whole thing